### PR TITLE
GUI/utility/libanimation/91

### PR DESCRIPTION
**Pull request by osvegn**
It implements the class Animation. This class is going to be used to create animation in the GUI client when entities move.
Be careful about the code clarity, the clarity of the documentation and the correct use of English language, and don't be afraid to suggest some feature to add in this class.

### CODE REVIEW

File: src/zappy_gui_src/Core/Animation/Animation.hpp [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+#include <SFML/Graphics.hpp>
+
+namespace gui {
+    /// \brief Animation class should be used to create an animation in sfml.
+    /// \note It contains a list of texture that will be updated, depending on a duration time, with the function update.
+    /// You should define a duration between each frame.
+    /// You should define a position for the animation.
+    /// You should define a size for the animation.
+    /// You should add a texture for the animation.
+    /// \warning Be careful, a texture is always add at the end of the vector of texture.
```
Comment (pariciadra) [inline src/zappy_gui_src/Core/Animation/Animation.hpp:22]: warning and be careful means the same things

File: src/zappy_gui_src/Core/Animation/Animation.hpp [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+            /// \param texturePath The path to load the texture from.
+            /// \param position The position to get the texture from in the image loaded.
+            /// \param size The size to get the texture from in the image loaded.
+            /// \throw AnimationException will be thrown if the image cannot be loaded.
+            void addTexture(const std::string &texturePath, const sf::Vector2f &position, const sf::Vector2f &size);
+
+            /// \brief It adds a texture on the vector of texture used for the animation.
+            /// \warning Be carful, the texture is added at the end of the vector. Please add texture in the correct order.
+            /// \param image The image used to create the texture from.
+            /// \param position The position to get the texture from in the image loaded.
```
Comment (pariciadra) [inline src/zappy_gui_src/Core/Animation/Animation.hpp:45]: The position from which the texture of the loaded image is to be obtained. Would be more accurate

File: src/zappy_gui_src/Core/Animation/Animation.hpp [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+/// \file src/zappy_gui_src/Core/Animation/Animation.hpp
+
+#ifndef ANIMATION_HPP_
+#define ANIMATION_HPP_
+
+#include <SFML/Graphics.hpp>
+
+namespace gui {
+    /// \brief Animation class should be used to create an animation in sfml.
+    /// \note It contains a list of texture that will be updated, depending on a duration time, with the function update.
```
Comment (pariciadra) [inline src/zappy_gui_src/Core/Animation/Animation.hpp:17]: You say list but there is no list in private member, maybe you mean vector

**Review by pariciadra — APPROVED** [review]
Great work, documentation is clear, maybe can you sometimes use reverso or deepl to speak more suitable English but it's ok overall

File: src/zappy_gui_src/Core/Animation/Animation.hpp (hunk 1 above)
Comment (osvegn) [reply src/zappy_gui_src/Core/Animation/Animation.hpp:22]: \warning is a balise for the documentation generation, and be careful is the begging of the sentance

**Review by pariciadra — COMMENTED** [review]
Great work, documentation is clear, maybe can you sometimes use reverso or deepl to speak more suitable English but it's ok overall

File: src/zappy_gui_src/Core/Animation/Animation.hpp (hunk 1 above)
Comment (pariciadra) [reply src/zappy_gui_src/Core/Animation/Animation.hpp:22]: that's why be carefull isn't useful here

**Review by LucasTesnier — APPROVED** [review]
NTR!

**Review by GuyomT — APPROVED** [review]
NTR

**Pull request merged**